Sources/HealthCheck/System.swift

```diff
@@ -0,0 +1,49 @@
+import Vapor
+
+// MARK: System
+struct System {
+    // System components that should respond for proper health check
+    var components: [HealthComponent.Type]
+
+    // Response headers
+    let headers: HTTPHeaders = HTTPHeaders([
+        ("Content-Type", "application/health+json"),
+        ("Cache-Control", "max-age=3600"),
+        ("Connection", "close")
+    ])
+
+    init(_ components: [HealthComponent.Type]) {
+        self.components = components
+    }
+
+    func health(on request: Request) -> EventLoopFuture<Response> {
+        self.components.map { $0.healthCheck(on: request) }
+            .flatten(on: request)
+            .flatMap(to: Response.self) { results in
+                var responseStatus: HTTPStatus = .ok
+                var healthStatus: Health.Status = .pass
+
+                results.forEach { check in
+                    if check.status == Health.Status.warn.rawValue {
```

Comment:
`if check.status == .warn {`

Reply:
with a `for check in checks` you can `break` once you found the first warning. And combine it with the check for failures like so:
```swift
forLoop: for check in checks {
  switch check.status {
  case .warn: healthStatus = .warn
  case .fail: 
    healthStatus = .fail
    responseStatus = .internalServerError
    break forLoop
  default: break
  }
}
```
See https://www.hackingwithswift.com/example-code/language/how-to-break-out-of-multiple-loop-levels-using-labeled-statements